docs/GUIDE.md

```diff
@@ -0,0 +1,142 @@
+- [Register.scala](../circe-akka-serializer/src/main/scala/org/virtuslab/ash/circe/Register.scala) (helper object with `apply` method that needs to be used in order to register Codecs for serializable type)
+- [Registration.scala](../circe-akka-serializer/src/main/scala/org/virtuslab/ash/circe/Registration.scala) (`Register.apply` method returns an instance of type Registration. Registration instances are later used as registered codecs)
+
+Scaladocs from these files and a quick lecture of [Akka serialization guide](https://doc.akka.io/docs/akka/current/serialization.html) should be enough to use Circe Akka Serializer properly. Apart from implementing the Circe Akka Serializer in scala code, remember to configure your Serializer in Akka-specific configuration file (.conf file). Two mandatory configurations are `akka.actor.serializers` and `akka.actor.serialization-bindings` (example below):
+```editorconfig
+akka {
+  actor {
+    serializers {
+      circe-json = "org.example.ExampleSerializer"
+    }
+    serialization-bindings {
+      "org.example.MySerializable" = circe-json
+    }
+  }
+}
+```
+where `org.example.ExampleSerializer` is the FQCN of your Serializer which extends CirceAkkaSerializer and `org.example.MySerializable` is the FQCN of your top-level serializable type (trait / abstract class) that is extended by Messages / Events / States which `org.example.ExampleSerializer` should serialize.
+
+Also, there are few optional configurations that you can set if you want to change the default behavior of Circe Akka Serializer:
+- `org.virtuslab.ash.circe.verbose-debug-logging` = `on` / `off`
+
+`on` enables debug logs for each serialization / deserialization -> prints info about what has been serialized/deserialized and how much time it took. Default is `off`.<br><br>
+
+- `org.virtuslab.ash.circe.enable-missing-codecs-check` = `true` / `false`
+
+`true` enables additional check for possible missing codec registrations (more info about this problem in [README](../README.md)). This is in general checked by the [Codec Registration Checker Compiler Plugin](#codec-registration-checker-compiler-plugin-guide) during compilation, so the default is `false`. However, if you use only Circe Akka Serializer without compiler plugins - this check should be enabled. Be aware that enabling this check might have a slight negative impact on Serializer's performance.<br><br>
+
+- `org.virtuslab.ash.circe.compression.algorithm` = `gzip` / `off`
+
+`gzip` enables compression of payloads before serialization using the [Gzip](https://www.gnu.org/software/gzip) algorithm. Default is `off` (compression disabled). Choosing `gzip` altogether with setting proper `compress-larger-than` value (explained below) enables compression of payloads.<br><br>
```

Comment:
I'd rather link to the library that is used in ASH for that purpose, not to GNU gzip

Reply:
But the only thing related to Gzip we use in code is:
```
import java.util.zip.GZIPInputStream
import java.util.zip.GZIPOutputStream
```
So, there is no good way to provide a link for these two types from `java.util.zip` package. What do you think?

Reply:
Maybe link to https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/zip/package-summary.html then?

Reply:
Hm, to be honest - that would not make sense to me as a reader. Let me jest remove the link then. People reading this guide would rather know about Gzip or just google for it.

Reply:
+1, this is a well-known software/algorithm, no need for link

Rel, fun fact: https://virtuslab.github.io/tips/#other/compression-zip-vs-targz